app/services/qa/linked_data/authority_url_service.rb

```diff
@@ -0,0 +1,62 @@
+# Provide service for constructing the external access URL for an authority.
+module Qa
+  module LinkedData
+    class AuthorityUrlService
+      # Build a url for an authority/subauthority for the specified action.
+      # @param authority [Symbol] name of a registered authority
+      # @param subauthority [String] name of a subauthority
+      # @param action [Symbol] action with valid values :search or :term
+      # @param action_request [String] the request the user is making of the authority (e.g. query text or term id/uri)
+      # @param substitutions [Hash] variable-value pairs to substitute into the URL template
+      # @returns a valid URL the submits the action request to the external authority
+      def self.build_url(action_config:, action:, action_request:, substitutions: {}, subauthority: nil)
+        # auth_config = Qa::Authorities::LinkedData::Config.new(authority)
+        action_validation(action)
+        # action_config = action_config(auth_config, action)
+        url_config = Qa::IriTemplate::UrlConfig.new(action_url(action_config, action))
+        selected_substitutions = url_config.extract_substitutions(substitutions)
+        selected_substitutions[action_request_variable(action_config, action)] = action_request
+        selected_substitutions[action_subauth_variable(action_config, action)] = action_subauth_variable_value(action_config, subauthority, action) if subauthority.present?
+
+        Qa::IriTemplateService.build_url(url_config: url_config, substitutions: selected_substitutions)
+      end
+
+      def self.action_validation(action)
+        return if [:search, :term].include? action
+        raise Qa::UnsupportedAction, "#{action} Not Supported - Action must be one of the supported actions (e.g. :term, :search)"
+      end
+      private_class_method :action_validation
```

Comment:
I think it is these lines that I find so hard to read.  It looks like there are a bunch of public class methods on the surface until you dig in.